src/__swaps__/screens/Swap/components/SwapNumberPad.tsx

```diff
@@ -49,23 +51,78 @@ const getFormattedInputKey = (inputKey: inputKeys) => {
+  const { nativeCurrency } = useAccountSettings();
+  const {
+    focusedInput,
+    internalSelectedInputAsset,
+    internalSelectedOutputAsset,
+    isQuoteStale,
+    SwapInputController,
+    configProgress,
+    outputQuotesAreDisabled,
+  } = useSwapContext();
 
   const longPressTimer = useSharedValue(0);
 
-  const addNumber = (number?: number) => {
+  const removeFormatting = (inputKey: inputKeys) => {
     'worklet';
+    return stripNonDecimalNumbers(SwapInputController[getFormattedInputKey(inputKey)].value);
+  };
+
+  const ignoreChange = ({ currentValue, addingDecimal = false }: { currentValue?: string; addingDecimal?: boolean }) => {
+    'worklet';
+    // ignore when: outputQuotesAreDisabled and we are updating the output amount or output native value
     if ((focusedInput.value === 'outputAmount' || focusedInput.value === 'outputNativeValue') && outputQuotesAreDisabled.value) {
+      return true;
+    }
+
+    // ignore when: corresponding asset does not have a price and we are updating native inputs
+    const inputAssetPrice = internalSelectedInputAsset.value?.nativePrice || internalSelectedInputAsset.value?.price?.value || 0;
+    const outputAssetPrice = internalSelectedOutputAsset.value?.nativePrice || internalSelectedOutputAsset.value?.price?.value || 0;
+    const outputAssetHasNoPrice = !outputAssetPrice || equalWorklet(outputAssetPrice, 0);
+    const inputAssetHasNoPrice = !inputAssetPrice || equalWorklet(inputAssetPrice, 0);
+    if (
+      (focusedInput.value === 'outputNativeValue' && outputAssetHasNoPrice) ||
+      (focusedInput.value === 'inputNativeValue' && inputAssetHasNoPrice)
+    ) {
+      return true;
+    }
+
+    // ignore when: decimals exceed native currency decimals
+    if (currentValue) {
+      const currentValueDecimals = currentValue.split('.')?.[1]?.length ?? -1;
+      const nativeCurrencyDecimals = supportedNativeCurrencies[nativeCurrency].decimals;
+
+      if (addingDecimal && nativeCurrencyDecimals === 0) {
+        return true;
+      } else if (
+        (focusedInput.value === 'inputNativeValue' || focusedInput.value === 'outputNativeValue') &&
+        currentValueDecimals >= nativeCurrencyDecimals
+      ) {
+        return true;
+      }
+    }
+    return false;
+  };
+
+  const addNumber = (number?: number) => {
+    'worklet';
+    const inputKey = focusedInput.value;
+    const currentValue = removeFormatting(inputKey);
+
+    if (ignoreChange({ currentValue })) {
       return;
     }
 
     // Immediately stop the quote fetching interval
     SwapInputController.quoteFetchingInterval.stop();
 
-    const inputKey = focusedInput.value;
-    const currentValue = stripCommas(SwapInputController[getFormattedInputKey(inputKey)].value);
+    const inputMethod = SwapInputController.inputMethod.value;
 
-    const newValue = currentValue === '0' ? `${number}` : `${currentValue}${number}`;
+    const isNativePlaceholderValue =
```

Comment:
need this to handle for the case when there is a placeholder zero value for the native inputs (e.g, "$0.00" that doesn't match "0")